src/test/java/ch/wisv/areafiftylan/products/service/TicketServiceTest.java

```diff
@@ -0,0 +1,664 @@
+        List<String> optionsString = Arrays.asList(CH_MEMBER_OPTION, PICKUP_SERVICE_OPTION);
+
+        Ticket ticket = ticketService.requestTicketOfType(user, typeString, optionsString);
+
+        assertEquals(user, ticket.getOwner());
+        assertEquals(optionSet, ticket.getEnabledOptions());
+        assertEquals(type, ticket.getType());
+    }
+
+    @Test
+    public void requestTicketOfTypeStringsTypeNotAvailable() {
+        User user = persistUser();
+        String typeString = "unavailable_ticket";
+        List<String> optionsString = Arrays.asList(CH_MEMBER_OPTION, PICKUP_SERVICE_OPTION);
+
+        thrown.expect(TicketTypeNotFoundException.class);
+        thrown.expectMessage("TicketType ");
+
+        ticketService.requestTicketOfType(user, typeString, optionsString);
+    }
+
+    @Test
+    public void requestTicketOfTypeStringsTypeNull() {
+        User user = persistUser();
+        String typeString = null;
+        List<String> optionsString = Arrays.asList(CH_MEMBER_OPTION, PICKUP_SERVICE_OPTION);
+
+        thrown.expect(TicketTypeNotFoundException.class);
+        thrown.expectMessage("TicketType ");
+
+        ticketService.requestTicketOfType(user, typeString, optionsString);
+    }
+
+    @Test
+    public void requestTicketOfTypeStringsOptionsNotFound() {
+        User user = persistUser();
+        String typeString = TEST_TICKET;
+        List<String> optionsString = Arrays.asList("not found", "nope.avi");
+
+        thrown.expect(TicketOptionNotFoundException.class);
+        thrown.expectMessage("Ticket Option not found!");
+
+        ticketService.requestTicketOfType(user, typeString, optionsString);
+    }
+
+    @Test
+    public void requestTicketOfTypeStringsOptionsNull() {
+        User user = persistUser();
+        TicketType type = ticketTypeRepository.findByName(TEST_TICKET).get();
+        Set<TicketOption> optionSet = new HashSet<>();
+        String typeString = TEST_TICKET;
+        List<String> optionsString = null;
+
+        Ticket ticket = ticketService.requestTicketOfType(user, typeString, optionsString);
+
+        assertEquals(user, ticket.getOwner());
+        assertEquals(optionSet, ticket.getEnabledOptions());
+        assertEquals(type, ticket.getType());
+    }
+
+    @Test
+    public void requestTicketOfTypeObjects() {
+        User user = persistUser();
+        TicketType type = ticketTypeRepository.findByName(TEST_TICKET).get();
+        Ticket ticketToPersist = new Ticket(user, type);
+        TicketOption pickupOption = ticketOptionRepository.findByName(PICKUP_SERVICE_OPTION).get();
+        ticketToPersist.addOption(pickupOption);
+        TicketOption chMemberOption = ticketOptionRepository.findByName(CH_MEMBER_OPTION).get();
+        ticketToPersist.addOption(chMemberOption);
+        List<TicketOption> options = Arrays.asList(pickupOption, chMemberOption);
+        Set<TicketOption> optionSet = new HashSet<>(options);
+        testEntityManager.persist(ticketToPersist);
```

Comment:
Because the requesting is what we test. The test then asserts if the retrieved `ticket` is equal to the one we persisted. How else could we test this method?